### PR TITLE
feat: Add patch to configure containerd metrics

### PR DESCRIPTION
**Pull request by dlipovetsky**

### CODE REVIEW

File: pkg/handlers/generic/mutation/containerdmetrics/inject_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/capi/clustertopology/handlers/mutation"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/testutils/capitest"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/testutils/capitest/request"
+)
+
+func TestContainerdRestartPatch(t *testing.T) {
```
Comment (supershal) [inline pkg/handlers/generic/mutation/containerdmetrics/inject_test.go:18]:
```suggestion
func TestContainerdMetricsPatch(t *testing.T) {
```

File: pkg/handlers/generic/mutation/containerdmetrics/inject_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	gomega.RegisterFailHandler(Fail)
+	RunSpecs(t, "Containerd restart mutator suite")
```
Comment (supershal) [inline pkg/handlers/generic/mutation/containerdmetrics/inject_test.go:20]:
```suggestion
	RunSpecs(t, "Containerd metrics mutator suite")
```

File: pkg/handlers/generic/mutation/containerdmetrics/inject_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+var _ = Describe("Generate Containerd restart patches", func() {
```
Comment (supershal) [inline pkg/handlers/generic/mutation/containerdmetrics/inject_test.go:23]:
```suggestion
var _ = Describe("Generate Containerd metrics patches", func() {
```

**Review by supershal — APPROVED** [review]
few nits. LGTM

**Review by dkoshkin — APPROVED** [review]

File: pkg/handlers/generic/mutation/containerdmetrics/inject_test.go (hunk 2 above)
Comment (dlipovetsky) [reply pkg/handlers/generic/mutation/containerdmetrics/inject_test.go:20]: Thanks, fixed

File: pkg/handlers/generic/mutation/containerdmetrics/inject_test.go (hunk 3 above)
Comment (dlipovetsky) [reply pkg/handlers/generic/mutation/containerdmetrics/inject_test.go:23]: Thanks, fixed

File: pkg/handlers/generic/mutation/containerdmetrics/inject_test.go (hunk 1 above)
Comment (dlipovetsky) [reply pkg/handlers/generic/mutation/containerdmetrics/inject_test.go:18]: Thanks, fixed

**Pull request merged**